File: benchmarks/comparison/fib.cpp
#include <iostream>
#include <chrono>

long long fib(long long n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

int main() {
    std::cout << "C++: Starting Fib(35)..." << std::endl;
    auto start = std::chrono::high_resolution_clock::now();
    long long res = fib(35);
    auto end = std::chrono::high_resolution_clock::now();
    auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - start);
    
    std::cout << "C++: Result = " << res << std::endl;
    std::cout << "C++: Time = " << duration.count() << "ms" << std::endl;
    return 0;
}

File: benchmarks/comparison/fib.rs
use std::time::Instant;

fn fib(n: i64) -> i64 {
    if n < 2 { return n; }
    fib(n - 1) + fib(n - 2)
}

fn main() {
    println!("Rust: Starting Fib(35)...");
    let start = Instant::now();
    let res = fib(35);
    let duration = start.elapsed();
    println!("Rust: Result = {}", res);
    println!("Rust: Time = {}ms", duration.as_millis());
}
